### ISSUE TITLE
💡 [REQUEST] - Extend Elasticsearch backend to be enable to use Opensearch

### ISSUE THREAD

**Issue by maximepln** [google/slo-generator#346]
### Summary

Since January 2021 and Elasticsearch re-licensing, the project isn't open source anymore. **OpenSearch** is a fork of Elasticsearch made in response of elastic becoming private.

As it stands, the Elasticsearch library is restricting it's usage to ElasticSearch server:
`UnsupportedProductError: The client noticed that the server is not Elasticsearch and we do not support this unknown product`

The idea will be to create another backend, or if code is similar enough to extend the actual Elasticsearch backend to be able to use Elasticsearch or Opensearch server. 

### Basic Example

Copy/paste from samples/elasticsearch/slo_elk_test_ratio.yaml

```apiVersion: sre.google.com/v2
kind: ServiceLevelObjective
metadata:
  name: elk-test-errors
  labels:
    service_name: elk
    feature_name: test
    slo_name: errors
spec:
  description: >
    SLO for random test data generated with the
    https://github.com/oliver006/elasticsearch-test-data
  backend: opensearch
  method: good_bad_ratio
  exporters:
  - pubsub
  - cloud_monitoring/test
  service_level_indicator:
    index: test_data
    date_field: last_updated
    query_good: {}
    query_bad:
      must:
        term:
          name: JAgOZE8
  goal: 1


### Screenshots

_No response_

### Drawbacks

None identified.



### Unresolved questions

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lvaylet** [comment]
Excellent idea. Let's move forward with this.

**Comment by lvaylet** [comment]
Closed by #348
